options/index.js

```diff
@@ -1,46 +1,60 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import ReactDOM from 'react-dom'
 
-ReactDOM.render(<h1>Hello!</h1>, document.getElementById('root'))
-
-chrome.management.getSelf((self) => {
-    if (self.installType === 'development') {
-        const hmrConnection = new WebSocket('ws://localhost:8080')
-
-        hmrConnection.onmessage = () => window.location.reload()
-    }
-})
-
-const webhookUrlField = document.getElementById('webhookUrlField')
-
-chrome.storage.sync.get('webhookUrl', (data) => setWebhookInputField(data))
-document.getElementById('saveButton').addEventListener('click', storeWebhookUrl)
-
-function setWebhookInputField(data) {
-    console.log(`Setting: ${data.webhookUrl}`)
-    webhookUrlField.value = data.webhookUrl
+import '../common/hot-reload'
+
+const Form = () => {
+    const [webhookUrl, setWebhookUrl] = useState('')
+
+    useEffect(() => {
+        chrome.storage.sync.get('webhookUrl', ({ webhookUrl }) =>
```

Comment:
What are the extra curly braces `{ }` here for? Isn't this just a lambda function that handles the result from `chrome.storage.sync.get`? Or is this some JSX thing that I'm not familiar with?
```suggestion
        chrome.storage.sync.get('webhookUrl', (webhookUrl) => setWebhookUrl(webhookUrl)
```

Reply:
Yeah good q, it's just JavaScript syntactic sugar, [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). 

That callback in `chrome.storage.sync.get` is getting a data object as an argument. The destructuring neatly allows you to pull the value from a field in one line.